engines/advancedDetector.cpp

```diff
@@ -626,6 +651,54 @@ static bool getFilePropertiesIntern(uint md5Bytes, const AdvancedMetaEngine::Fil
 	return true;
 }
 
+// Add backslash before double quotes (") and backslashes themselves (\)
+Common::String escapeString(const char *string) {
+	if (string == nullptr || Common::String(string) == "")
+		return "NULL";
+
+	Common::String res = "";
+
+	for (int i = 0; string[i] != '\0'; i++) {
+		if (string[i] == '"' || string[i] == '\\')
+			res += "\\";
+
+		res += string[i];
+	}
+
+	return res;
+}
+
+void AdvancedMetaEngineDetection::dumpDetectionEntries() const {
+	const byte *descPtr;
+
+	for (descPtr = _gameDescriptors; ((const ADGameDescription *)descPtr)->gameId != nullptr; descPtr += _descItemSize) {
+		auto g = ((const ADGameDescription *)descPtr);
+		const char *title = ((const PlainGameDescriptor *)_gameIds)->description;
+
+		debug("game (");
+		debug("\tname %s", escapeString(g->gameId).c_str());
```

Comment:
Aren't you supposed to put these escaped strings into quotes?

Reply:
The quotes are so that we can account for strings with spaces in them. For metadata, the parser only sees the first whitespace, so this is not necessary.